consul/src/main/scala/io/buoyant/consul/CatalogNamer.scala

```diff
@@ -2,16 +2,21 @@ package io.buoyant.consul
 
 import com.twitter.finagle._
 import com.twitter.util._
-import io.buoyant.consul.v1.{UnexpectedResponse, ServiceNode}
+import io.buoyant.consul.v1.{Indexed, UnexpectedResponse, ServiceNode}
 
 class CatalogNamer(
   idPrefix: Path,
-  api: v1.CatalogApi,
-  includeTag: Boolean = false
+  catalogApi: v1.CatalogApi,
+  agentApi: v1.AgentApi,
+  includeTag: Boolean = false,
+  setHost: Boolean = false
 ) extends Namer {
 
   import CatalogNamer._
 
+  lazy val domainFuture: Future[String] =
+    agentApi.self().map(_.Config.get.Domain.get.stripPrefix(".").stripSuffix("."))
```

Comment:
According to Consul API spec the `Domain` field cannot be empty and thus `Future.exception(???)` should never be executed. Would be OK to leave it as is or I should replace `???` with some concrete exception (and which one would be appropriate in such case)?